### PR TITLE
[systemtest] Refactor OLM related classes

### PR DESCRIPTION
**Pull request by im-konge**
### Type of change

- Refactoring

### Description

This PR refactors OLM installation method inside our system tests. 

It also:

- adds dependency on fabric8's `openshift-model-operatorhub`, which adds resources like `Subscription`, `OperatorGroup`, `ClusterServiceVersion` etc.
- adds `OlmConfiguration` class, which is used for easier installation and handling of all parameters inside the `OlmResource` class.
- adds supporting classes like `SubscriptionResource` and `OperatorGroupResource` for easier handling using `ResourceManager`
- removes all of the methods, which were using `executor` for getting resources like `Subscription`, `OperatorGroup` and `ClusterServiceVersion`
- removes all the yaml files related to OLM

### Checklist

- [ ] Make sure all tests pass

### CODE REVIEW

File: systemtest/pom.xml [hunk 1]
```diff
@@ -234,6 +234,15 @@
...
         </dependency>
+        <dependency>
+            <groupId>io.fabric8</groupId>
+            <artifactId>openshift-model-operatorhub</artifactId>
+            <version>${fabric8.openshift-client.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>io.fabric8</groupId>
+            <artifactId>openshift-client-api</artifactId>
+        </dependency>
```
Comment (scholzj) [inline systemtest/pom.xml:245]: You should probably use the dependency management in the main `pom.xml` and not add them only here.

File: test/pom.xml [hunk 2]
```diff
@@ -87,6 +87,11 @@
             <artifactId>kubernetes-model-admissionregistration</artifactId>
             <version>${fabric8.kubernetes-model.version}</version>
         </dependency>
+        <dependency>
+            <groupId>io.fabric8</groupId>
+            <artifactId>openshift-model-operatorhub</artifactId>
+            <version>${fabric8.openshift-client.version}</version>
+        </dependency>
```
Comment (scholzj) [inline test/pom.xml:94]: Same as above.

**Comment by jankalinic** [comment]
LGTM

**Review by jankalinic — APPROVED** [review]

**Review by see-quick — APPROVED** [review]

**Review by scholzj — APPROVED** [review]
LGTM I guess (or at least the comments I had were addressed)

**Comment by im-konge** [comment]
OLM upgrade:
image

**Comment by im-konge** [comment]
Here is the acceptance - please note that the failures are connected to external listeners, which we cannot run on the usual OCP cluster and also to the flakiness of the tests (will be fixed in next PR):

image

The OLM installation works without an issue.

**Review by Frawless — APPROVED** [review]

**Pull request merged**